### PR TITLE
Replace telebot with aiogram

### LINKED ISSUE #127: Another Telegram API library for bot
The current library, [pyTelegramBotAPI](https://github.com/eternnoir/pyTelegramBotAPI), lacks functionality, e.g. support for middleware that runs after message handler or convenient exception handling mechanism.

Candidates for replacement (in order of the personal preference):
* [aiogram](https://github.com/aiogram/aiogram)
  * **a**synchronous  
  * ~~mature~~ acceptable middleware processing
  * [proper offset update](https://github.com/aiogram/aiogram/blob/1e2fe72aca12a3fc6f2d1f66c71539af5a84ea00/aiogram/dispatcher/dispatcher.py#L369) no matter what handlers will do
* [python-telegram-bot](https://github.com/python-telegram-bot/python-telegram-bot)
  * synchronous 
  *  ~~_no middleware?_~~ some kind of indirect middleware with [`TypeHandler`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.typehandler.html?highlight=TypeHandler#telegram.ext.TypeHandler)
  * has periodic jobs system (although, we'll need that not only for the bot, either way – see #129)
  * ~~disk persistence (_no need for tgbot application?_)~~ it's better to delegate persistence-related stuff to the database
* [pyrogram](https://github.com/pyrogram/pyrogram)
  * **a**synchronous or synchronous 
  * no middleware
* [telethon](https://github.com/LonamiWebs/Telethon)

### PR DESCRIPTION
**Pull request by TmLev**
This pull request aims to:
* Replace obsolete `telebot` and synchronous `requests` with modern `aiogram` and asynchronous `aiohttp`.
* Provide convenient wrappers for API calls.
* Reimplement basic bot commands: `/set_code`, `/my_code`, `/reset_code`
* Add `AuthMiddleware` for filtering out requests from unauthorized users and logging for debugging.
* Configure YAPF for `tgbot`.

Should resolve #127.

### CODE REVIEW

**Comment by TmLev** [comment]
Ready for review. @gakhromov

File: back-end/Pipfile [hunk 1]
```diff
@@ -26,4 +26,4 @@ psycopg2-binary = "*"
 uvicorn = "*"
 
 [requires]
-python_version = "3.8.5"
+python_version = "3.9.1"
```
Comment (gakhromov) [inline back-end/Pipfile:29]: finally

File: tgbot/src/api/auth.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+Session = namedtuple("Session", ["id", "code", "chat_id"])
+
+
+async def fetch_session(params: tp.Dict[str, tp.Any]) -> tp.Optional[Session]:
+    """Fetch session for chat. Session may not exist, hence `Optional`."""
+
+    async with client.get("session", params=params) as response:
+        data: tp.List[tp.Dict[str, tp.Any]] = await response.json()
+
```
Comment (gakhromov) [inline tgbot/src/api/auth.py:19]: In Pyhton 3.9, we can use `params: dict[...]` instead of `params: tp.Dict[...]` (PEP: 585)

Comment (gakhromov) [reply tgbot/src/api/auth.py:19]: The same applies for lists btw

File: tgbot/src/api/auth.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    """
+
+    session = await fetch_session(params={"chat_id": chat_id})
+    assert session is not None
+
+    return session.code
+
+
+async def patch_session(id_: int, data: tp.Dict[str, tp.Any]) -> ClientResponse:
+    """Patch some fields of session."""
```
Comment (gakhromov) [inline tgbot/src/api/auth.py:44]: Same PEP: 585 stuff

File: tgbot/src/api/auth.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+from api.client import client
+
+# ------------------------------------------------------------------------------
+
+Session = namedtuple("Session", ["id", "code", "chat_id"])
+
+
+async def fetch_session(params: tp.Dict[str, tp.Any]) -> tp.Optional[Session]:
+    """Fetch session for chat. Session may not exist, hence `Optional`."""
+
```
Comment (gakhromov) [inline tgbot/src/api/auth.py:16]: I would like to name this function `get_session`, but I'm still ok with `fetch` version. Is there some sort of a naming rule that I do not know?

File: tgbot/src/api/auth.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    return DeauthorizeResult(success=True, details="")
+
+
+# ------------------------------------------------------------------------------
+
+User = namedtuple("User", ["full_name", "platoon"])
+
+
+async def fetch_user(chat_id: int) -> User:
+    return User(full_name="ФИО", platoon="1809")
```
Comment (gakhromov) [inline tgbot/src/api/auth.py:98]: `platoon` might be a better name for groups, but we are kinda used to `milgroup`s, so, please, change it to `milgroup`s

File: tgbot/src/handlers/auth.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    user = await fetch_user(chat_id)
+    await message.reply(f"Здравия желаю, {user.full_name}!\n"
+                        f"Должность: командир взвода {user.platoon}")
+
+
+async def my_code(message: Message) -> None:
+    # `fetch_code` contract is fulfilled by auth middleware
+    code = await fetch_code(message.chat.id)
+    await message.reply(f"Текущий код: {md_code(code)}", parse_mode=MD2)
+
```
Comment (gakhromov) [inline tgbot/src/handlers/auth.py:44]: Cool thing with MD, I like that

**Review by gakhromov — CHANGES_REQUESTED** [review]
Very nice! I like the new code very much, streets ahead of `pyTelegramAPI` junk. Check out my comments, I think something should be slightly changed

File: tgbot/src/api/auth.py (hunk 4 above)
Comment (TmLev) [reply tgbot/src/api/auth.py:16]: `fetch` implies network access.

Comment (gakhromov) [reply tgbot/src/api/auth.py:16]: Ok

File: tgbot/src/api/auth.py (hunk 2 above)
Comment (TmLev) [reply tgbot/src/api/auth.py]: Fixed.

File: tgbot/src/api/auth.py (hunk 3 above)
Comment (TmLev) [reply tgbot/src/api/auth.py]: Fixed.

File: tgbot/src/api/auth.py (hunk 5 above)
Comment (TmLev) [reply tgbot/src/api/auth.py]: Renamed.

**Review by gakhromov — APPROVED** [review]
LGTM.

**Pull request merged**